src/iso19111/c_api.cpp

```diff
@@ -2674,6 +2674,40 @@ PROJ_STRING_LIST proj_get_codes_from_database(PJ_CONTEXT *ctx,
 
 // ---------------------------------------------------------------------------
 
+/** \brief Return the list of celestial body codes used in the database.
+ *
+ * The returned list is NULL terminated and must be freed with
+ * proj_string_list_destroy().
+ *
+ * @param ctx PROJ context, or NULL for default context
+ * @param auth_name Authority name (must not be NULL)
```

Comment:
you could potentially allow NULL to be passed in to allow querying in all authorities. See proj_create_from_name()